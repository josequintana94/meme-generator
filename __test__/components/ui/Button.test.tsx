import renderer from "react-test-renderer";
import Button from "../../../src/components/ui/Button";

const component = renderer.create(
  <Button
    id="Button"
    type="submit"
    onClick={jest.fn()}                
  > 
    PROCEED TO CHECKOUT
  </Button>            
);

describe(">>>> Button", () => {
  test("++++ Snapshot Button", () => {   
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Button component", () => {
  test("it shows the expected text when clicked (testing the wrong way!)", () => {   
    const instance = component.root;
    const button = instance.findByType("button");
    button.props.onClick();
    expect(button.props.children).toBe("PROCEED TO CHECKOUT");
  });
});

export {};