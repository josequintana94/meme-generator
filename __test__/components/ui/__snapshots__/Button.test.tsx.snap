// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> Button ++++ Snapshot Button 1`] = `
<button
  id="Button"
  onClick={[MockFunction]}
  type="submit"
>
  PROCEED TO CHECKOUT
</button>
`;
